src/components/tooltip/tooltip.test.js

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/button-has-type */
+import * as React from "react";
+import { forwardRef } from "react";
+// import Button from "../button";
+import Tooltip from "./tooltip.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+import { tooltipPreview } from "../../../cypress/locators/tooltip/index";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+const TooltipComponent = ({ ...props }) => {
+  // const [position, setPosition] = useState("top");
+  const Component = forwardRef(({ children }, ref) => (
+    <button
+      tabIndex="0"
+      style={{
+        backgroundColor: "#00815D",
+        color: "white",
+        cursor: "pointer",
+        border: "none",
+        padding: "8px",
+      }}
+      ref={ref}
+    >
+      {children}
+    </button>
+  ));
+  return (
+    <>
+      <div
+        style={{
+          display: "flex",
+        }}
+      >
+        {/* <Button onClick={() => setPosition("top")}>Top Position</Button>
+            <Button onClick={() => setPosition("bottom")}>
+              Bottom Position
+            </Button>
+            <Button onClick={() => setPosition("left")}>Left Position</Button>
+            <Button onClick={() => setPosition("right")}>Right Position</Button> */}
+      </div>
+      <div
+        style={{
+          padding: "60px 60px 60px 160px",
+        }}
+      >
+        <Tooltip {...props} message="I am a tooltip!" isVisible>
+          <Component>target</Component>
+        </Tooltip>
+      </div>
+    </>
+  );
+};
+
+context("Tests for Tooltip component", () => {
+  describe("should check Tooltip component properties", () => {
+    it.each(testData)(
+      "should check %s as message for Tooltip component",
+      (message) => {
+        CypressMountWithProviders(<Tooltip isVisible message={message} />);
```

Comment:
you are passing `isVisible` as default to the component, why do we need to duplicate this here ?

Reply:
I'm not using custom component here so I need to pass isVisible as I'm passing message prop in custom component so not able to test with testdata using < TooltipComponent >